File: XMLCheck/Handler/CheckRulesXML15/Xml15RuleRunner.cs
using XMLCheck.Handler.CheckRulesXML04;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML15
{
    public static class Xml15RuleRunner
    {
        private static readonly List<IXml15Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(List<Xml15Model>? models,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            if (models == null || models.Count == 0)
                return errors;

            foreach (var model in models)
            {
                foreach (var rule in _rules)
                {
                    var err = rule.Check(model,context);
                    if (err != null)
                        errors.Add(err);
                }
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml15/xml15RuleRunner.ts
import { Xml15Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml15Rule } from './xml15RuleBase';

export class Xml15RuleRunner {
  private static readonly rules: IXml15Rule[] = [];

  public static run(models: Xml15Model[], context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!models || models.length === 0) return errors;

    for (const model of models) {
      for (const rule of this.rules) {
        const err = rule.check(model, context);
        if (err) {
          errors.push(err);
        }
      }
    }

    return errors;
  }
}
